daemon/algod/api/server/v2/handlers.go

```diff
@@ -781,6 +787,23 @@ func (v2 *Handlers) GetStatus(ctx echo.Context) error {
 		CatchpointAcquiredBlocks:    &stat.CatchpointCatchupAcquiredBlocks,
 	}
 
+	upgradePropose := string(stat.UpgradePropose)
+	votesToGo := uint64(stat.NextProtocolVoteBefore) - uint64(stat.LastRound)
+	if upgradePropose != "" && votesToGo > 0 {
+		consensus := config.Consensus[protocol.ConsensusCurrentVersion]
+		upgradeVoteRounds := consensus.UpgradeVoteRounds
+		upgradeThreshold := consensus.UpgradeThreshold
+		votes := uint64(upgradeVoteRounds) - votesToGo
+		votesYes := stat.NextProtocolApprovals
+		votesNo := votes - votesYes
+		upgradeDelay := uint64(stat.UpgradeDelay)
+		response.UpgradePropose = &upgradePropose
+		response.UpgradeThreshold = &upgradeThreshold
+		response.UpgradeApprove = &latestBlkHdr.UpgradeApprove
+		response.UpgradeDelay = &upgradeDelay
+		response.UpgradeNo = &votesNo
+	}
```

Comment:
All of this can go in `Node.Status`, then you can unconditionally initialize the response (and avoid looking up the block header)

Reply:
done